go/apps/api/routes/v2_analytics_get_verifications/handler.go

```diff
@@ -0,0 +1,257 @@
+		for _, api := range apis {
+			keySpaceIds = append(keySpaceIds, api.KeyAuthID)
+		}
+
+		securityFilters = append(securityFilters, chquery.SecurityFilter{
+			Column:        "key_space_id",
+			AllowedValues: keySpaceIds,
+		})
+	}
+
+	parser := chquery.NewParser(chquery.Config{
+		WorkspaceID:     auth.AuthorizedWorkspaceID,
+		Limit:           int(settings.MaxQueryResultRows),
+		SecurityFilters: securityFilters,
+		TableAliases: map[string]string{
+			"key_verifications_v1":            "default.key_verifications_raw_v2",
+			"key_verifications_per_minute_v1": "default.key_verifications_per_minute_v2",
+			"key_verifications_per_hour_v1":   "default.key_verifications_per_hour_v2",
+			"key_verifications_per_day_v1":    "default.key_verifications_per_day_v2",
+			"key_verifications_per_month_v1":  "default.key_verifications_per_month_v2",
+		},
+		AllowedTables: []string{
+			"default.key_verifications_raw_v2",
+			"default.key_verifications_per_minute_v2",
+			"default.key_verifications_per_hour_v2",
+			"default.key_verifications_per_day_v2",
+			"default.key_verifications_per_month_v2",
+		},
+	})
+
+	parsedQuery, err := parser.Parse(ctx, req.Query)
+	if err != nil {
+		return fault.Wrap(err,
+			fault.Code(codes.App.Validation.InvalidInput.URN()),
+			fault.Public("Invalid SQL query"),
+		)
+	}
+
+	// Now we build permission checks based on the key_space_id(s) one specified in the query itself
+	// If none are specified, we will just check if there is wildcard read_analytics permission set
+	permissionChecks := []rbac.PermissionQuery{
+		// Wildcard API analytics access
+		rbac.T(rbac.Tuple{
+			ResourceType: rbac.Api,
+			ResourceID:   "*",
+			Action:       rbac.ReadAnalytics,
+		}),
+	}
+
+	keySpaceIds := parser.ExtractColumn("key_space_id")
+	if len(keySpaceIds) > 0 {
+		keySpaces, keySpaceHits, err := h.Caches.KeyAuthToApiRow.SWRMany(
+			ctx,
+			array.Map(keySpaceIds, func(keySpaceId string) cache.ScopedKey {
+				return cache.ScopedKey{WorkspaceID: auth.AuthorizedWorkspaceID, Key: keySpaceId}
+			}),
+			func(ctx context.Context, keys []cache.ScopedKey) (map[cache.ScopedKey]db.FindKeyAuthsByKeyAuthIdsRow, error) {
+				keySpaces, err := db.Query.FindKeyAuthsByKeyAuthIds(ctx, h.DB.RO(), db.FindKeyAuthsByKeyAuthIdsParams{
+					WorkspaceID: auth.AuthorizedWorkspaceID,
+					KeyAuthIds: array.Map(keys, func(keySpace cache.ScopedKey) string {
+						return keySpace.Key
+					}),
+				})
+
+				if err != nil {
+					return nil, err
+				}
+
+				return array.Reduce(
+					keySpaces,
+					func(acc map[cache.ScopedKey]db.FindKeyAuthsByKeyAuthIdsRow, api db.FindKeyAuthsByKeyAuthIdsRow) map[cache.ScopedKey]db.FindKeyAuthsByKeyAuthIdsRow {
+						acc[cache.ScopedKey{WorkspaceID: auth.AuthorizedWorkspaceID, Key: api.KeyAuthID}] = api
+						return acc
+					},
+					map[cache.ScopedKey]db.FindKeyAuthsByKeyAuthIdsRow{},
+				), nil
+			},
+			caches.DefaultFindFirstOp,
+		)
+		if err != nil {
+			return err
+		}
```

Comment:
Nit but not nit 😄 . This block is quite hard to digest and read. I think we should definitely break that into smaller chunks I can't even tell where the block begins and ends. There is also one more above.